### DISCUSSION TITLE
What determines if a pixel is 'valid' for `/cog/statistics`

### DISCUSSION THREAD

**Discussion by mmnoo** [developmentseed/titiler#711] [Q&A] [answered]
Does the `/cog/statistics` endpoint only consider pixel values that are completely within an area of interest?

If yes, it might be nice to be able send in a parameter to instruct the calculations to account for pixel values that intersect the edge of the area of interest.

**Comment by vincentsarago** [comment, marked as answer]
@mmnoo depends on your titiler version but starting with version [`0.15.0`](https://github.com/developmentseed/titiler/blob/main/CHANGES.md#0150-2023-09-28) the `/cog/statistics [POST]` endpoint will return statistics weighted with the spatial coverage of the intersecting shape.

> only consider pixel values that are completely within an area of interest?

By default rio-tiler will **includes** all pixels that touches the shape, but as I said in titiler we've added spatial weight. 
https://github.com/cogeotiff/rio-tiler/blob/775ecfd513b452765154c20f9693db764e6cf69f/rio_tiler/io/rasterio.py#L573-L581

<img width="600" alt="Screenshot 2023-10-13 at 9 25 03 PM" src="https://github.com/developmentseed/titiler/assets/10407788/d572b1c9-7d55-405a-8990-6870d38cb531">

We could maybe make this configurable but I'm not sure I have time right now 

ref: 
- https://github.com/cogeotiff/rio-tiler/pull/598#pullrequestreview-1398469061
- https://github.com/cogeotiff/rio-tiler/pull/598#issuecomment-1520410993
- https://github.com/cogeotiff/rio-tiler/pull/598#pullrequestreview-1403765410
- https://github.com/cogeotiff/rio-tiler/pull/598#pullrequestreview-1404163542
